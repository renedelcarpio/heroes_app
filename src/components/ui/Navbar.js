import { Link, NavLink } from 'react-router-dom';

import React from 'react';

export const Navbar = () => {
	return (
		<nav className='navbar'>
			<Link className='navbar__brand' to='/'>
				Heroes App
			</Link>

			<div className='navbar__collapse'>
				<div className='navbar__nav'>
					<NavLink
						activeClassName='active'
						className='navbar__nav-item'
						exact
						to='/marvel'
					>
						Marvel
					</NavLink>

					<NavLink
						activeClassName='active'
						className='navbar__nav-item'
						exact
						to='/dc'
					>
						DC
					</NavLink>
				</div>
			</div>

			<div className='navbar__collapse'>
				<ul className='navbar__nav'>
					<NavLink
						activeClassName='active'
						className='navbar__nav-item'
						exact
						to='/login'
					>
						Logout
					</NavLink>
				</ul>
			</div>
		</nav>
	);
};
