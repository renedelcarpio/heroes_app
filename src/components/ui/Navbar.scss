@import '../../index.scss';

.navbar {
  background-color: $color-dark;
  display: flex;
  justify-content: space-around;
  align-items: center;
  height: 5rem;
  width: 100vw;

  .navbar__brand {
    color: $color-light;
    font-size: 2rem;
    font-weight: 700;
    text-decoration: none;
    width: 20%;

    &:hover {
      color: $color-secondary-marvel;
    }
  }

  &__heroes {
    display: flex;
    justify-content: space-around;
    width: 20%;
  }

  &__login {
    width: 30%;
  }

  &__nav {
    display: flex;
    align-items: center;
  }

  &__nav-item {
    color: $color-light;
    font-size: 1.5rem;
    font-weight: bold;
    margin-left: 1.5rem;
    text-decoration: none;
    transition: all 0.3s;

    &:hover {
      color: $color-mid;
    }
  }
}

.marvel-active {
  color: $color-primary-marvel;
  font-size: 1.5rem;
    font-weight: bold;
    margin-left: 1.5rem;
    text-decoration: none;
    transition: all 0.3s;
}

.dc-active {
  color: $color-primary-dc;
  font-size: 1.5rem;
    font-weight: bold;
    margin-left: 1.5rem;
    text-decoration: none;
    transition: all 0.3s;

    &:hover {
      font-size: 1.8rem;
    }
}

.search-active {
  color: $color-secondary-marvel;
  font-size: 1.5rem;
    font-weight: bold;
    margin-left: 1.5rem;
    text-decoration: none;
    transition: all 0.3s;

    &:hover {
      font-size: 1.8rem;
    }
}

.button {
  background-color: transparent;
  border: 0.2rem solid $color-mid;
  border-radius: 0.5rem;
  font-size: 1.3rem;
  font-weight: 400;
  padding: 0.3rem 0.6rem;

  &:hover {
    border: 0.2rem solid $color-secondary-dc;
    color: $color-secondary-dc;
  }
}

@media screen and(max-width: 400px) {
  .name-button {
    display: flex;
    flex-direction: column;
  }
}
