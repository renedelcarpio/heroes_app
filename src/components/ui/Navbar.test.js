import { AuthContext } from '../../auth/authContext';
import { MemoryRouter } from 'react-router-dom';
import { Navbar } from './Navbar';
import { mount } from 'enzyme';
import { types } from '../../types/types';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockNavigate,
}));

describe('Tests in Navbar component', () => {
	const contextValue = {
		dispatch: jest.fn(),
		user: {
			logged: true,
			name: 'Pedro',
		},
	};

	const wrapper = mount(
		<AuthContext.Provider value={contextValue}>
			<MemoryRouter initialEntries={['/']}>
				<Navbar />
			</MemoryRouter>
		</AuthContext.Provider>
	);

	test('should match with snapshot', () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.user-name').text().trim()).toBe('Pedro');
	});

	test('should call logout, call navigate and call dispatch', () => {
		wrapper.find('button').prop('onClick')();

		expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });
		expect(mockNavigate).toHaveBeenCalledWith('/login', { replace: true });
	});
});
