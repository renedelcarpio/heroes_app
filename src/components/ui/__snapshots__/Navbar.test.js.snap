// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Navbar component should match with snapshot 1`] = `
<nav
  className="navbar"
>
  <a
    className="navbar__brand"
    href="/"
    onClick={[Function]}
  >
    Heroes App
  </a>
  <div
    className="navbar__heroes"
  >
    <div
      className="navbar__nav"
    >
      <a
        className="navbar__nav-item "
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="navbar__nav-item "
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="navbar__nav-item "
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar__login"
  >
    <ul
      className="navbar__nav name-button"
    >
      <span
        className="navbar__nav-item user-name"
      >
        Pedro
      </span>
      <button
        className="navbar__nav-item button"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
